codexctl/device.py

```diff
@@ -438,18 +572,49 @@ def restore_previous_version(self) -> None:
 /sbin/fw_setenv "fallback_partition" "${OLDPART}"
 /sbin/fw_setenv "active_partition" "${NEWPART}\""""
 
-        if self.hardware == HardwareType.RMPP:
-            RESTORE_CODE = """#!/bin/bash
-OLDPART=$(< /sys/devices/platform/lpgpr/root_part)
-if [[ $OLDPART  ==  "a" ]]; then
-    NEWPART="b"
-else
-    NEWPART="a"
-fi
-echo "new: ${NEWPART}"
-echo "fallback: ${OLDPART}"
-echo $NEWPART > /sys/devices/platform/lpgpr/root_part
-"""
+        if self.hardware in (HardwareType.RMPP, HardwareType.RMPPM):
+            _, _, current_version, _, backup_version = self.get_device_status()
+            current_part, inactive_part, _ = self._get_paper_pro_partition_info(current_version)
+
+            new_part_label = "a" if inactive_part == 2 else "b"
+
+            parts = current_version.split('.')
+            if len(parts) >= 2 and parts[0].isdigit() and parts[1].isdigit():
+                current_is_new = [int(parts[0]), int(parts[1])] >= [3, 22]
+            else:
+                raise SystemError(f"Cannot restore: unexpected current version format '{current_version}'")

```

Comment:
```suggestion
            if len(parts) < 2 or not parts[0].isdigit() or not parts[1].isdigit():
                raise SystemError(f"Cannot restore: unexpected current version format '{current_version}'")

            current_is_new = [int(parts[0]), int(parts[1])] >= [3, 22]
```
Early exit allows you to avoid else statements which reduces clutter.